webapp/src/components/CreateMeetingNoteForm.tsx

```diff
@@ -0,0 +1,112 @@
+import { Alert, Card, CardContent, Snackbar, TextField } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import React, { FormEventHandler, useState, useContext } from 'react';
+
+import { EntityId } from '../types/api';
+import SessionContext from './SessionContext';
+
+interface CreateMeetingNoteFormProps {
+  meetingId: EntityId;
+  onMeetingNoteChanged?: (id: EntityId) => void;
+  greatestSortOrderOfMeetingNotesList: number;
+}
+
+const CreateMeetingNoteForm = ({
+  meetingId,
+  onMeetingNoteChanged,
+  greatestSortOrderOfMeetingNotesList,
+}: CreateMeetingNoteFormProps) => {
+  const { principalId } = useContext(SessionContext);
+
+  const [isSavingMeetingNote, setIsSavingMeetingNote] = useState(false);
+  const [saveMeetingNoteError, setSaveMeetingNoteError] = useState(null);
+  const [isSuccessMessageVisible, setIsSuccessMessageVisible] = useState(false);
+  const [meetingNoteText, setMeetingNoteText] = useState('');
+  const onSubmit: FormEventHandler = event => {
+    event.preventDefault();
+    setIsSavingMeetingNote(true);
+
+    fetch(`${process.env.REACT_APP_API_ORIGIN}/meetings/${meetingId}/notes`, {
+      method: 'POST',
+      credentials: 'include',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        agenda_owning_participant_id: principalId,
+        note_text: meetingNoteText,
+        sort_order: greatestSortOrderOfMeetingNotesList + 1,
+      }),
+    })
+      .then(res => res.json())
+      .then(({ data, error }) => {
+        setIsSavingMeetingNote(false);
+        if (error) {
+          setSaveMeetingNoteError(error);
+        }
+        if (data) {
+          setIsSuccessMessageVisible(true);
+          setMeetingNoteText('');
+          if (onMeetingNoteChanged) {
+            onMeetingNoteChanged(data.id);
+          }
+        }
+      })
+      .catch(error => {
+        setIsSavingMeetingNote(false);
+        setSaveMeetingNoteError(error);
+      });
+  };
+  return (
+    <form onSubmit={onSubmit}>
+      <Card>
+        <CardContent sx={{ pb: 0 }}>
+          <h3>Add Meeting Note</h3>
```

Comment:
Which `<h2>` does this fall under? Should this even be a heading?

Reply:
So, if it should be a heading, why an `<h3>` nested under an unknown or absent `<h2>`? Headings ought to be nested correctly, and it appears as though this on assumes where the component will be mounted will result in correct nesting.

Reply:
(Also, would an improved label be enough? It seems strange to base this on `CreateOrUpdateCompetencyForm` since the two forms have dramatically different use cases.)

Reply:
> Just looking at the screenshot:
> 
> * Why isn't there any margin between the meeting notes and the form?
> * Why is the form in a card, even though the sidebar doesn't have very much space for the extra padding?
> * Why doesn't the text input span the full width of its container?
> * Why is there so much space between the text input and the button?

These issues have been addressed by reviewing the product spec and following those style guidelines.

Reply:
The heading tags were removed to avoid any problems with incorrect nesting.